subsys/bluetooth/audio/ascs.c

```diff
@@ -934,14 +935,18 @@ static void ase_release(struct bt_ascs_ase *ase)
 	}
 
 	if (unicast_server_cb != NULL && unicast_server_cb->release != NULL) {
-		err = unicast_server_cb->release(ase->ep.stream);
+		err = unicast_server_cb->release(ase->ep.stream, &rsp);
 	} else {
 		err = -ENOTSUP;
 	}
 
-	if (err) {
-		ascs_cp_rsp_add_errno(ASE_ID(ase), BT_ASCS_RELEASE_OP, err,
-				      BT_ASCS_REASON_NONE);
+	if (err || rsp.code != BT_ASCS_RSP_SUCCESS) {
```

Comment:
Isn't it only if `err != 0` that the `rsp.code` should be set? 

Shouldn't this just be
```suggestion
	if (err) {
```

And then if `rsp.code == BT_ASCS_RSP_SUCCESS`, then it means that the upper layer did not correctly fill out the response code, in which case we should set it to `BT_ASCS_RSP_UNSPECIFIED`